### PR TITLE
[IMPROVEMENT] Added More Unit tests

### PR DESCRIPTION
**Pull request by T1duS**
Please prefix your pull request with one of the following: **[FEATURE]** **[FIX]** **[IMPROVEMENT]**.

**In raising this pull request, I confirm the following (please check boxes):**

- [X] I have read and understood the [contributors guide](https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [X] I have checked that another pull request for this purpose does not exist.
- [X] I have considered, and confirmed that this submission will be valuable to others.
- [X] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [X] I give this submission freely, and claim no ownership to its content.

**My familiarity with the project is as follows (check one):**

- [ ] I have never used the project.
- [ ] I have used the project briefly.
- [ ] I have used the project extensively, but have not contributed previously.
- [X] I am an active contributor to the project.

---

I have added some unit tests to increase codecov coverage.

### CODE REVIEW

File: tests/test_auth/TestControllers.py [hunk 1]
```diff
@@ -117,3 +118,59 @@ def test_github_token_validity(self):
...
+            response = c.post(
+                '/account/login', data=self.create_login_form_data(self.user.email, self.user.password))
+            response = c.post(
+                '/account/manage',data=dict(
+                    current_password=self.user.password,
+                    name="T1duS",
+                    email=self.user.email
+                ))
+            user = User.query.filter(User.name == "T1duS").first()
+            self.assertNotEqual(user, None)
```
Comment (saurabhshri) [inline tests/test_auth/TestControllers.py]: If the user's name is edited successfully, why is this returning None? Also, could you please also assert for testing  if the success message is in response?

File: tests/test_auth/TestControllers.py [hunk 2]
```diff
@@ -117,3 +118,59 @@ def test_github_token_validity(self):
...
+            response = c.post(
+                '/account/login', data=self.create_login_form_data(self.user.email, self.user.password))
+            response = c.post(
+                '/account/manage',data=dict(
+                    current_password=self.user.password,
+                    name=self.user.name,
+                    email="valid@gmail.com"
+                ))
+            user = User.query.filter(User.email == "valid@gmail.com").first()
+            self.assertNotEqual(user, None)
```
Comment (saurabhshri) [inline tests/test_auth/TestControllers.py]: Same, if email is edited, why is it none?

File: tests/test_auth/TestControllers.py [hunk 3]
```diff
@@ -117,3 +118,59 @@ def test_github_token_validity(self):
...
+            response = c.post(
+                '/account/login', data=self.create_login_form_data(self.user.email, self.user.password))
+            response = c.post(
+                '/account/manage',data=dict(
+                    current_password=self.user.password,
+                    name=self.user.name,
+                    email="invalid@gg"
+                ))
+            user = User.query.filter(User.email == "invalid@gg").first()
+            self.assertEqual(user, None)
```
Comment (saurabhshri) [inline tests/test_auth/TestControllers.py]: How is this different from the test above `test_edit_email` apart from diff email?

File: tests/test_auth/TestControllers.py (hunk 1 above)
Comment (T1duS) [reply tests/test_auth/TestControllers.py]: It is assert**Not**Equal

File: tests/test_auth/TestControllers.py (hunk 2 above)
Comment (T1duS) [reply tests/test_auth/TestControllers.py]: assert**Not**Equal

File: tests/test_auth/TestControllers.py (hunk 3 above)
Comment (T1duS) [reply tests/test_auth/TestControllers.py]: this is assertEqual, the other was assertNotEqual

File: tests/test_auth/TestControllers.py (hunk 1 above)
Comment (saurabhshri) [reply tests/test_auth/TestControllers.py]: @T1duS Makes sense. Sorry, I misread it. Still would like to have assert for testing if the success message is in response?

File: tests/test_auth/TestControllers.py (hunk 3 above)
Comment (saurabhshri) [reply tests/test_auth/TestControllers.py]: @T1duS Makes sense. Sorry, I misread it. Still would like to have assert for testing if the failure message is in response?

File: tests/test_auth/TestControllers.py [hunk 4]
```diff
@@ -117,3 +118,63 @@ def test_github_token_validity(self):
         """
         res = github_token_validity('token')
         self.assertEqual(res, False)
+
+class ManageAccount(BaseTestCase):
+
+    def test_edit_username(self):
+        """
+        Tests if username is edited
```
Comment (saurabhshri) [inline tests/test_auth/TestControllers.py]:
```suggestion
        Test editing user's name.
```

File: tests/test_auth/TestControllers.py [hunk 5]
```diff
@@ -117,3 +118,63 @@ def test_github_token_validity(self):
...
+                    name="T1duS",
+                    email=self.user.email
+                ))
+            user = User.query.filter(User.name == "T1duS").first()
+            self.assertNotEqual(user, None)
+            self.assertIn("Settings saved", str(response.data))
+
+    def test_edit_email(self):
+        """
+        Tests if user email is edited
```
Comment (saurabhshri) [inline tests/test_auth/TestControllers.py]:
```suggestion
        Test editing user's email address.
```

File: tests/test_auth/TestControllers.py [hunk 6]
```diff
@@ -117,3 +118,63 @@ def test_github_token_validity(self):
...
+                    name=self.user.name,
+                    email="valid@gmail.com"
+                ))
+            user = User.query.filter(User.email == "valid@gmail.com").first()
+            self.assertNotEqual(user, None)
+            self.assertIn("Settings saved", str(response.data))
+
+    def test_edit_invalid_email(self):
+        """
+        Test for editing an invalid email
```
Comment (saurabhshri) [inline tests/test_auth/TestControllers.py]:
```suggestion
        Test editing user's email with invalid email address.
```

File: tests/test_ci/TestControllers.py [hunk 7]
```diff
@@ -202,4 +203,94 @@ def test_inform_mailing_list(self, mock_email):
...
                 'subject': 'GitHub Issue #matejmecka', 'to': 'ccextractor-dev@googlegroups.com'
             }
-)
\ No newline at end of file
+        )
+
+
+    def test_add_blocked_users(self):
+        """
+        Check it will add blocked user
```
Comment (saurabhshri) [inline tests/test_ci/TestControllers.py]:
```suggestion
        Check adding a user to block list.
```

File: tests/test_ci/TestControllers.py [hunk 8]
```diff
@@ -202,4 +203,94 @@ def test_inform_mailing_list(self, mock_email):
...
+        """
+        self.create_user_with_role(
+            self.user.name, self.user.email, self.user.password, Role.admin)
+        with self.app.test_client() as c:
+            response = c.post(
+                '/account/login', data=self.create_login_form_data(self.user.email, self.user.password))
+            response = c.post(
+                '/blocked_users', data=dict(user_id=1, comment="Bad user", add=True))
+            self.assertNotEqual(BlockedUsers.query.filter(BlockedUsers.comment=="Bad user").first(),None)
```
Comment (saurabhshri) [inline tests/test_ci/TestControllers.py]: Better to fetch from blocked users list, the user with id=1 and see if it's not none. Also add other asserts to check the comment and any success message in response.

File: tests/test_ci/TestControllers.py [hunk 9]
```diff
@@ -202,4 +203,94 @@ def test_inform_mailing_list(self, mock_email):
...
+
+    def test_add_blocked_users_wrong_id(self):
+        """
+        Check it will not add blocked users with wrong id
```
Comment (saurabhshri) [inline tests/test_ci/TestControllers.py]:
```suggestion
        Check adding invalid user id to block list.
```

File: tests/test_ci/TestControllers.py [hunk 10]
```diff
@@ -202,4 +203,94 @@ def test_inform_mailing_list(self, mock_email):
...
+        """
+        self.create_user_with_role(
+            self.user.name, self.user.email, self.user.password, Role.admin)
+        with self.app.test_client() as c:
+            response = c.post(
+                '/account/login', data=self.create_login_form_data(self.user.email, self.user.password))
+            response = c.post(
+                '/blocked_users', data=dict(user_id=0, comment="Bad user", add=True))
+            self.assertEqual(BlockedUsers.query.filter(BlockedUsers.comment=="Bad user").first(),None)
```
Comment (saurabhshri) [inline tests/test_ci/TestControllers.py]: Check the if response has the error message.

File: tests/test_ci/TestControllers.py [hunk 11]
```diff
@@ -202,4 +203,94 @@ def test_inform_mailing_list(self, mock_email):
...
+        Check it will not add blocked users with empty id
+        """
+        self.create_user_with_role(
+            self.user.name, self.user.email, self.user.password, Role.admin)
+        with self.app.test_client() as c:
+            response = c.post(
+                '/account/login', data=self.create_login_form_data(self.user.email, self.user.password))
+            response = c.post(
+                '/blocked_users', data=dict(comment="Bad user", add=True))
+            self.assertEqual(BlockedUsers.query.filter(BlockedUsers.comment=="Bad user").first(),None)
```
Comment (saurabhshri) [inline tests/test_ci/TestControllers.py]: There could be another blocked user with this comment. Remove this and check the error message in response and if any blocked user has null id.

File: tests/test_ci/TestControllers.py [hunk 12]
```diff
@@ -202,4 +203,94 @@ def test_inform_mailing_list(self, mock_email):
...
+        with self.app.test_client() as c:
+            response = c.post(
+                '/account/login', data=self.create_login_form_data(self.user.email, self.user.password))
+            response = c.post(
+                '/blocked_users', data=dict(user_id=0, comment="Bad user", add=True))
+            self.assertEqual(BlockedUsers.query.filter(BlockedUsers.comment=="Bad user").first(),None)
+
+    def test_add_blocked_users_empty_id(self):
+        """
+        Check it will not add blocked users with empty id
```
Comment (saurabhshri) [inline tests/test_ci/TestControllers.py]:
```suggestion
        Check adding blank user id to block list.
```

File: tests/test_ci/TestControllers.py [hunk 13]
```diff
@@ -202,4 +203,94 @@ def test_inform_mailing_list(self, mock_email):
...
+        with self.app.test_client() as c:
+            response = c.post(
+                '/account/login', data=self.create_login_form_data(self.user.email, self.user.password))
+            response = c.post(
+                '/blocked_users', data=dict(comment="Bad user", add=True))
+            self.assertEqual(BlockedUsers.query.filter(BlockedUsers.comment=="Bad user").first(),None)
+
+    def test_add_blocked_users_already_exists(self):
+        """
+        Check it will show message when an already blocked user is added to the blocked users list
```
Comment (saurabhshri) [inline tests/test_ci/TestControllers.py]:
```suggestion
        Check adding existing blocked user again.
```

File: tests/test_ci/TestControllers.py [hunk 14]
```diff
@@ -202,4 +203,94 @@ def test_inform_mailing_list(self, mock_email):
...
+            g.db.commit()
+            response = c.post(
+                '/blocked_users', data=dict(user_id=1, comment="Bad user", add=True))
+            with c.session_transaction() as session:
+                flash_message = dict(session['_flashes']).get('message')
+            self.assertEqual(flash_message,"User already blocked.")            
+
+    def test_remove_blocked_users(self):
+        """
+        Check it will remove blocked users
```
Comment (saurabhshri) [inline tests/test_ci/TestControllers.py]:
```suggestion
        Check removing user from block list.
```

File: tests/test_ci/TestControllers.py [hunk 15]
```diff
@@ -202,4 +203,94 @@ def test_inform_mailing_list(self, mock_email):
...
+        with self.app.test_client() as c:
+            response = c.post(
+                '/account/login', data=self.create_login_form_data(self.user.email, self.user.password))
+            blocked_user = BlockedUsers(1,"Bad user")
+            g.db.add(blocked_user)
+            g.db.commit()
+            self.assertNotEqual(BlockedUsers.query.filter(BlockedUsers.comment=="Bad user").first(),None)
+            response = c.post(
+                '/blocked_users', data=dict(user_id=1, remove=True))
+            self.assertEqual(BlockedUsers.query.filter(BlockedUsers.comment=="Bad user").first(),None)
```
Comment (saurabhshri) [inline tests/test_ci/TestControllers.py]: Check ID instead of comment.

File: tests/test_ci/TestControllers.py [hunk 16]
```diff
@@ -202,4 +203,94 @@ def test_inform_mailing_list(self, mock_email):
...
+
+    def test_remove_blocked_users_wrong_id(self):
+        """
+        Check it will not remove blocked users with wrong id
```
Comment (saurabhshri) [inline tests/test_ci/TestControllers.py]:
```suggestion
        Check removing non existing id from block list.
```

File: tests/base.py [hunk 17]
```diff
@@ -244,13 +244,14 @@ def complete_forktest(self, test_id, regression_tests):
...
         g.db.commit()
 
-    def create_user_with_role(self, user, email, password, role):
+    def create_user_with_role(self, user, email, password, role, github_token=None):
         """
         Create a user with specified user details and role.
         """
         from flask import g
         user = User(self.user.name, email=self.user.email,
                     password=User.generate_hash(self.user.password), role=role)
```
Comment (satyammittal) [inline tests/base.py]: you can pass github_token here itself

File: tests/base.py [hunk 18]
```diff
@@ -57,11 +57,13 @@ def load_config(file):
             }
 
 
-def MockRequests(url, data=None):
+def MockRequests(url, data=None, timeout=None):
     if url == "https://api.github.com/repos/test/test_repo/commits/abcdef":
-        return MockResponse({}, 200)
+        return MockResponse({}, 200)
```
Comment (satyammittal) [inline tests/base.py]: remove extra space.

**Pull request merged**